apps/explorer/src/components/top-validators-card/TopValidatorsCard.tsx

```diff
@@ -50,91 +86,139 @@ export const STATE_DEFAULT: ValidatorState = {
-            })
-            .catch((error: any) => {
-                console.log(error);
-                setLoadState('fail');
-            });
-    }, [rpc]);
-
-    if (loadState === 'loaded') {
-        return <TopValidatorsCard state={showObjectState as ValidatorState} />;
-    }
-    if (loadState === 'pending') {
-        return (
-            <div data-testid="validators-table">
-                <TabGroup>
-                    <TabList>
-                        <Tab>Validators</Tab>
-                    </TabList>
-                    <TabPanels>
-                        <TabPanel>
-                            <div title="Top Validators">
-                                <PlaceholderTable
-                                    rowCount={3}
-                                    rowHeight="13px"
-                                    colHeadings={[
-                                        'Name',
-                                        'Address',
-                                        'Pubkey Bytes',
-                                    ]}
-                                    colWidths={['135px', '220px', '220px']}
-                                />
-                            </div>
-                        </TabPanel>
-                    </TabPanels>
-                </TabGroup>
-            </div>
-        );
-    }
-    if (loadState === 'fail') {
-        return <ValidatorLoadFail />;
-    }
+function stakeColumn(validator: {
+    stake: BigInt;
+    stakePercent: number;
+}): JSX.Element {
+    return (
+        <div className="flex gap-1.5 items-end">
+            <Text variant="bodySmall" color="steel-darker">
+                {validator.stake.toString()}
+            </Text>
+            <Text variant="caption" color="steel">
+                {validator.stakePercent.toFixed(2)}%
+            </Text>
+        </div>
+    );
+}
 
-    return <div>Something went wrong</div>;
+export function processValidators(set: Validator[], totalStake: bigint) {
+    return set
+        .map((av) => {
+            const rawName = av.fields.metadata.fields.name;
+            const name = textDecoder.decode(
+                new Base64DataBuffer(rawName).getData()
+            );
+            return {
+                name: name,
+                address: av.fields.metadata.fields.sui_address,
+                pubkeyBytes: av.fields.metadata.fields.pubkey_bytes,
+                stake: av.fields.stake_amount,
+                stakePercent: getStakePercent(
+                    av.fields.stake_amount,
+                    totalStake
+                ),
+                delegation_count: av.fields.delegation_count || 0,
+            };
+        })
+        .sort((a, b) => (a.name > b.name ? 1 : -1));
 }
 
-function TopValidatorsCard({ state }: { state: ValidatorState }) {
-    const validatorsData = processValidators(
-        state.validators.fields.active_validators
+export const getStakePercent = (stake: bigint, total: bigint): number =>
```

Comment:
The number conversion is lossy, we should use the big number library we use to avoid that.

Reply:
done.